db/revision_cache_interface.go

```diff
@@ -260,9 +263,42 @@ func revCacheLoaderForDocument(backingStore RevisionCacheBackingStore, doc *Docu
 	history = encodeRevisions(validatedHistory)
 	channels = doc.History[revid].Channels
 
+	// If we have "_attachments" present in the raw body (pre-2.5 metadata), we'll need an unmarshalled version of the body.
 	if doc.CurrentRev == revid {
 		body = doc._body
+		attachments = doc.Attachments
+	}
+
+	if bytes.Contains(bodyBytes, []byte(`"`+BodyAttachments+`"`)) && body == nil {
+		if err := body.Unmarshal(bodyBytes); err != nil {
+			return nil, nil, nil, nil, nil, false, &time.Time{}, err
+		}
+	}
+
+	if pre25Atts, pre25Body, hasPre25Attachment, err := extractPre25Attachments(body); err != nil {
+		return nil, nil, nil, nil, nil, false, &time.Time{}, err
+	} else if hasPre25Attachment {
+		attachments = pre25Atts
```

Comment:
If attachments are already non-empty (from doc.Attachments), should do a merge to be defensive.  

In the case that an attachment exists in both, use the one with higher revpos.